tangelo/problem_decomposition/dmet/_helpers/dmet_scf_guess.py

```diff
@@ -55,3 +54,64 @@ def dmet_fragment_guess(t_list, bath_orb, chemical_potential, norb_high, number_
     frag_guess = np.dot(eigenvectors[ :, : int(number_active_electron/2)], eigenvectors[ :, : int(number_active_electron/2)].T) * 2
 
     return frag_guess
+
+
+def dmet_fragment_guess_rohf_uhf(t_list, bath_orb, chemical_potential, norb_high,
+    number_active_electron, active_fock_alpha, active_fock_beta, n_active_alpha,
+     n_active_beta):
+    """Construct the guess orbitals.
+
+    Args:
+        t_list (list): Number of fragment & bath orbitals (int).
+        bath_orb (numpy.array): The bath orbitals (float64).
+        chemical_potential (float64): The chemical potential.
+        norb_high (int): The number of orbitals in the fragment calculation.
+        number_active_electron (int): The number of electrons in the fragment
+            calculation.
+        active_fock_alpha (numpy.array): The fock matrix from the low-level
+            calculation for the alpha electrons (float64).
+        active_fock_beta (numpy.array): The fock matrix from the low-level
+            calculation for the beta electrons (float64).
+        number_active_electron (int): The number of electrons in the fragment
+            calculation.
+        n_active_alpha (int): The number octive alpha electrons.
+        n_active_beta (int): The number octive beta electrons.
+
+    Returns:
+        frag_guess (numpy.array): The guess orbitals (float64).
+    """
+
+    n_spin = n_active_alpha - n_active_beta
+    n_pair = (number_active_electron - n_spin) // 2
+    new_alpha = n_pair + n_spin
+    new_beta = n_pair
+
+    norb = t_list[0]
+
+    # Construct the fock matrix of the fragment (subtract the chemical potential for consistency)
+    fock_fragment = bath_orb[:, : norb_high].T @ active_fock_alpha @ bath_orb[:, : norb_high]
+    if(chemical_potential != 0):
+        for i in range(norb):
+            fock_fragment[i, i] -= chemical_potential
+
+    # Diagonalize the fock matrix and get the eigenvectors
+    eigenvalues, eigenvectors = scipy.linalg.eigh(fock_fragment)
+    eigenvectors = eigenvectors[:, eigenvalues.argsort()]
+
+    # Extract the eigenvectors of the occupied orbitals as the guess orbitals
+    # Introduce alpha- and beta-electrons
+    frag_guess_alpha = np.dot(eigenvectors[:, :int(new_alpha)], eigenvectors[:, :int(new_alpha)].T)
+
+    # Construct the fock matrix of the fragment (subtract the chemical potential for consistency)
+    fock_fragment = bath_orb[:, :norb_high].T @ active_fock_beta @ bath_orb[:, :norb_high]
+    if(chemical_potential != 0):
```

Comment:
If `chemical_potential` is 0., we perform `norb` unnecessary operations (we subtract 0).

You just performed a bunch of matrix multiplications  (complexity `n^3`) and even a matrix diagonalization (`n^4`?). I fail to see how these `if` lines are helping us. My first reaction was to think it was chemistry related rather than compute-saving. Am I missing the point ?

I do not understand why we are distracting the reader and complexifying code if there is nothing to gain, unless I am missing something. My first reaction was to think the `if` was chemistry-related rather than compute-saving. Am I missing the point ?

Reply:
From my understanding, it is just to avoid doing the loop if the chemical potential is 0.